lua/neo-tree/ui/renderer.lua

```diff
@@ -543,6 +553,10 @@ local draw = function(nodes, state, parent_id)
     M.set_expanded_nodes(state.tree, expanded_nodes)
   end
   state.tree:render()
+
+  -- Restore the cursor position/focused node in the tree based on the state
+  -- when it was last closed
+  state.position.restore()
```

Comment:
If this works in all situations, it's an elegant solution. I want to test it locally and try to break it before merging.